File: java/DS/Arrays/deleteElement.java


class deleteElement {
    //Function to search the key to be deleted
    static int findElement(int arr[], int n, int key){
        for(int i = 0; i < n; i++){
            if(arr[i] == key){
                return i;
            }
        }
        return -1;
    }
    static int deleteElement(int arr[], int n, int key){
        //find position of element to be deleted
        int pos = findElement(arr, n, key);

        if (pos == -1){
            System.out.println("Element not found.");
            return n;
        }

        for(int i = pos; i < n - 1; i++){
            arr[i] = arr[i + 1];
        }
        return n - 1;
    }

    public static void main(String[] args) {
        int i;
        int arr[] = {10, 50, 30, 40, 20};

        int n = arr.length;
        int key = 30;

        System.out.println("Array before deletion");
        for(i = 0; i < n; i++){
            System.out.println(arr[i] + " ");
        }
        //function call
        n = deleteElement(arr, n, key);

        System.out.println("\n\nArray after deletion");
        for(i = 0; i < n; i++){
            System.out.println(arr[i] + " ");
        }
    }
}

File: c/ds/deleteElement.c
#include<stdio.h>

void deleteElement(int array[], int *size, int position);
int main(void){
    int capacity = 4;
    int arr[capacity];
    arr[0] = 1;
    arr[1] = 2;
    arr[2] = 3;
    int n = 3;
    int pos = 2; 

    for(int i = 0; i < n; i++){
        printf("%d ", arr[i]);
    }
    printf("\n");

    deleteElement(arr, &n, pos);

    for(int i = 0; i < n; i++){
        printf("%d ", arr[i]);
    }
    printf("\n");

}

void deleteElement(int array[], int*size, int position){
    if((position >= *size) || (position < 0)){
        printf("Invalid position");
    }else{
        for(int i = position; i <= (*size - 2); i++){
            array[i] = array[i+1];
        }
        *size-=1;
    }
}